sdk/python/touca/cli/_plugin.py

```diff
@@ -62,17 +63,27 @@ def __init__(self, options: dict):
 
     def _command_list(self):
         for member in user_plugins():
-            print("{}: {}".format(member.name, member.help))
+            print("{}: {}".format(member.name, member.help))   
         return True
 
     def _command_add(self):
         from shutil import copyfile
-
         plugin_name = self.__options.get("name")
         plugin_path_src = Path.cwd().joinpath(plugin_name).with_suffix(".py")
         if not plugin_path_src.exists():
-            print(f'did not find a plugin with name "{plugin_name}"', file=sys.stderr)
-            return False
+            try:
+                res = requests.get(plugin_name)
+                plugin_path_dst = Path(find_home_path(), "plugins", os.path.basename(plugin_name)).with_suffix(".py")
+                if plugin_path_dst.exists():
+                    print(f'plugin "{plugin_name}" is already installed', file=sys.stderr)
+                    return False
+                with open(plugin_path_dst, 'wb') as f:
+                    f.write(res.content)
+                return True
+            #request threw error beacause plugin does not exist
+            except:
+                print(f'did not find a plugin with name "{plugin_name}"', file=sys.stderr)
+                return False
```

Comment:
I don't think this change very well described to the reader what we're doing and how the logic fits together. It also borrows logic from before/after.

What do you think about simplifying the impl for this whole function to the following?
```python
    def _command_add(self):
        from shutil import copyfile
        from urllib.parse import urlparse

        plugin_arg: str = self.__options.get("name")
        plugin_name = plugin_arg
        if plugin_arg.startswith("https://"):
            plugin_name = Path(urlparse(plugin_arg).path).stem
        plugin_path_dst = Path(find_home_path(), "plugins", plugin_name).with_suffix(
            ".py"
        )
        plugin_path_dst.parent.mkdir(exist_ok=True)
        if plugin_path_dst.exists():
            print(f'plugin "{plugin_name}" is already installed', file=sys.stderr)
            return False
        plugin_path_src = Path.cwd().joinpath(plugin_name).with_suffix(".py")
        if plugin_path_src.exists():
            copyfile(plugin_path_src, plugin_path_dst)
            return True
        res = requests.get(plugin_arg)
        if res.status_code == 200:
            plugin_path_dst.write_bytes(res.content)
            return True
        print(f'did not find a plugin at "{plugin_arg}"', file=sys.stderr)
        return False
```

Reply:
I like this implementation more. I say you just merge this version and I take this as a learning experience in writing concise code

Reply:
Thank you for your graciousness, Ryan! Will do. Just trying to improve the readability so we can maintain this code easier.